site/content/blog/2020-05-04-pritunl.md

```diff
@@ -0,0 +1,175 @@
+---
+title: "From OpenVPN to Pritunl VPN: The transition"
+slug: pritunl
+date: 2020-05-04T12:00:00-04:00
+author: Angelos Kyratzakos
+github: angeloskyratzakos
+community: angelos.kyratzakos
+---
+
+// JW note: I think a small section here or in the below paragraph talking about the importance of a good and reliable VPN when running a SaaS service while working remotely would be good. Gives people who aren't familiar with VPNs some background on why this was important for us to do
+
+Many of us we are using VPN to access internal-private resources of the infrastructure. A majority of which is using OpenVPN as a solution to host those VPN connections on their own servers. [OpenVPN](https://en.wikipedia.org/wiki/OpenVPN) is a widely used software which was also our selection to be used as a quickly and reliable solution to access our internal infrastructure. After using it a while we needed a better solution in terms of:
+- High Availability (HA)
+- Auditing
+- Supporting better access control (e.g. only SREs can access production servers and developers to access development account)
+- Working with SSO, particularly OneLogin
+After some investigation of the team, we ended up with [Pritunl](https://pritunl.com/) as it covered our needs and its pricing was reasonably for our scale.
+
+// JW note: Maybe a small section here talking about Pritunl a little more. e.g. a bit about the company, the product, why we liked it, etc.
+
+After some investigation of the team, we ended up with [Pritunl](https://pritunl.com/) as it covered our needs and its pricing was reasonably for our scale.
+
+#### Pritunl - Infrastructure
+To deploy Pritunl in our infrastructure, we used [Terraform](https://www.terraform.io/). The module we wrote can be found [here](https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/terraform/aws/modules/pritunl) as an example for how to deploy Pritunl. The infrastructure consists of:
+- 1 Route53 record,  
+- 1 Network Load Balancer, 
+- 2 AutoScaling Groups (ASG) and 
+- a MongoDB Atlas.
+
+{{< figure src="/blog/2020-05-04-pritunl/Pritunl_architecture.png" alt="Pritunl Architecture">}}
+
+##### Why 2 AutoScaling Groups?
+The selection of 2 ASGs with one instance has been done due to the nessesity of having the same Elastic Network Interface (ENI), which results on having the same private and public IPs. This is useful on whitelisting those IPs into the Security Groups that Pritunl-VPN needs to access internally such as our internal Gitlab. This can be achieved by attaching those ENIs as a secondary network interfaces on the instance, check appendix [Attach second ENI](#attach-second-eni)
```

Comment:
```suggestion
The selection of two ASGs with one instance has been done due to the necessity of having the same Elastic Network Interface (ENI), which results on having the same private and public IPs. This is useful on whitelisting those IPs into the Security Groups that Pritunl-VPN needs to access internally such as our internal GitLab. 

This can be achieved by attaching those ENIs as a secondary network interfaces on the instance, check appendix [Attach second ENI](#attach-second-eni).
```